### PR TITLE
[IA-4703] New Bioconductor and GATK images (with samtools 1.18)

### PR DESCRIPTION
**Pull request by LizBaldo**
__Jira ticket__: 
- https://broadworkbench.atlassian.net/browse/IA-4703
- https://broadworkbench.atlassian.net/browse/IA-4701

### What

- Sister PR to cache the new images implemented here https://github.com/DataBiosphere/terra-docker/pull/472

### Why

- New bioconductor
- Upgrade samtools from 1.10 to 1.18

### What to test

- 

### Who tested and where

- [ ] This change is covered by automated tests 
  - _NB: Rerun automation tests on this PR by commenting `jenkins retest` or `jenkins multi-test`._
- [ ] I validated this change 
- [ ] Primary reviewer validated this change 
- [ ] I validated this change in the dev environment

### CODE REVIEW

File: jenkins/dataproc-custom-images/prepare-custom-leonardo-jupyter-dataproc-image.sh [hunk 1]
```diff
@@ -16,15 +16,15 @@ set -e -x
...
-terra_jupyter_r="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-r:2.2.3"
-terra_jupyter_bioconductor="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-bioconductor:2.2.3"
+terra_jupyter_python="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-python:1.1.5"
+terra_jupyter_r="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-r:2.2.4"
+terra_jupyter_bioconductor="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-bioconductor:2.2.4"
 terra_jupyter_hail="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-hail:1.1.7"
-terra_jupyter_gatk="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-gatk:2.3.4"
-terra_jupyter_aou="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-aou:2.2.7"
+terra_jupyter_gatk="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-gatk:2.3.6"
+terra_jupyter_aou="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-aou:2.2.8"
```
Comment (LizBaldo) [inline jenkins/dataproc-custom-images/prepare-custom-leonardo-jupyter-dataproc-image.sh:24]: @Qi77Qi how would you like me to proceed with new AOU images given that your release process is a bit different than for Terra?

Also, can I remove the legacy code to support 2.1.15? I don't think you are using it anymore, correct?

Comment (LizBaldo) [reply jenkins/dataproc-custom-images/prepare-custom-leonardo-jupyter-dataproc-image.sh:24]: Update - I will not remove the legacy AOU support in this PR

**Review by Qi77Qi — APPROVED** [review]
assuming the versions are all intended...lgtm

**Comment by LizBaldo** [comment]
jenkins retest

**Comment by LizBaldo** [comment]
jenkins retest

**Comment by LizBaldo** [comment]
jenkins retest

**Comment by LizBaldo** [comment]
None of the integration tests are failing for the same reason, and they passed in previous commits so I am going to force merge now

**Pull request merged**